apps/api/src/api.controller.ts

```diff
@@ -17,4 +19,76 @@ export class ApiController {
       status: HttpStatus.OK,
     };
   }
+
+  @Put('asset/:userDsnpId/upload')
+  @UseInterceptors(FilesInterceptor('files'))
+  @HttpCode(202)
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({
+    description: 'Asset files',
+    type: FilesUploadDto,
+  })
+  // eslint-disable-next-line no-undef
+  async assetUpload(@Param('userDsnpId') userDsnpId: string, @UploadedFiles() files: Array<Express.Multer.File>): Promise<UploadResponseDto> {
```

Comment:
Do we care which user the uploaded content comes from?

As long as the actual broadcast/reply/etc... has it, we don't really care who it came from. (The outside application might for security reasons, but I think this service doesn't)

That said, it isn't a big deal either way.

Reply:
Had same question, do we care whose content is coming in anyways?

Reply:
so the content owner is the `userDsnpId` in this setup since it it uploading the that resource.

About capturing from which provider this request is coming from I can add another userId for submitter but there are no ways of validating that if we are not using any cookies or signatures and this is supposed to be ran inside a providers infrastructure then the answer to that question would be obvious.

 I'm open to any ideas. One other problem is that this is not supposed to be a persistent layer and DB that stores and retrieves all announcements, that seems to be the job of content indexer or something similar.

Reply:
Couldn't we just drop the user content owner? It's just unlinked content until used by the user on the other call right?

Reply:
you are correct, assets are not linked to any specific user until announcement submit

Reply:
removed extra route param